# Notes for next year

## For instance
* **Can't find it** remove gitk and and git_kraken
* **done** make /data read only
* **done (disabled)** remove screen saver
* **done** Fix swirl: `sudo chown -R ubuntu /usr/local/lib/R/site-library/swirl/`
* **done** install.packages('rsconnect') # instead of shinyapps
* **done** sudo apt-get install ruby-dev nodejs #needed for bundler, etc to display website locally
* **done** _but need to work on an alias_ install https://github.com/rajanil/fastStructure
* install trimmomatic
* install autobarcode
* **done** install star
* **done** IGV DOESN'T WORK.  DO IT RIGHT!!! (Get from website, not apt-get)
* **done (version v1.1.0-60-gc15b070 works)** Freebayes...currently have to use version 1.0.2
* installation sources should go in /usr/local/src not /usr/src
* **done** make screenshot work

## For Instance 2019

* fix whitespace issue in atom **reinstall atom...and if needed turn off whitespace package**
* R packages: learnr, tidyverse
* Replace /data with files that do not have multiple isoforms!!!


## For labs (now updating for 2019)

## Midterm

Be careful about blastx, blastp, etc

### Water Lab

Remove...

Or if keeping, redo "how long does it take" questions because they don't quite make sense now given how "part 5" is worded.


### BLAST Lab

Change paralog search to use a proteome file that does not have all the different isoforms

use for annotation of a genome

#### Rice SNPs

use separate template files for SNPs and GWAS.

#### Rice GWAS

Change to GAPIT

#### Illumina

Change to STAR

#### RNAseq GO terms

change the lab so that rownames are not used in the code; no one loads it in with rownames = geneID

provide the DE gene lists so that everyone starts with the same file.

#### Clustering

Is the gene expression normalized?  Doesn't seem like it...

#### Metagenomics

increase sample size?
Are the reps technical?  Prob not
Change the word "predominant" in the Qs


# 2018 notes for 2019

## Linux Lab

Change from Korf to http://linuxcommand.org/?
