# Notes for next year

## For instance
* **Can't find it** remove gitk and and git_kraken
* **appears to be autofixed in new version** fix whitespace issue in atom 
* **done** make /data read only
* **done (disabled)** remove screen saver
* **done** Fix swirl: `sudo chown -R ubuntu /usr/local/lib/R/site-library/swirl/`
* **done** install.packages('rsconnect') # instead of shinyapps
* **done** sudo apt-get install ruby-dev nodejs #needed for bundler, etc to display website locally
* **done** _but need to work on an alias_ install https://github.com/rajanil/fastStructure
* install trimmomatic
* install autobarcode
* install star
* IGV DOESN'T WORK.  DO IT RIGHT!!! (Get from website, not apt-get)
* Freebayes...currently have to use version 1.0.2
* installation sources should go in /usr/local/src not /usr/src
* make screenshot work

## For labs

### Water Lab

Remove

### BLAST Lab

Change paralog search to use a proteome file that does not have all the different isoforms

Extend the lab into R so that we find all orthologs

use for annotation of a genome

#### Rice SNPs

Change to use fastStructure instead of PSMix

#### Rice GWAS

Change to GAPIT

#### Illumina

Change to STAR

#### RNAseq GO terms

change the lab so that rownames are not used in the code; no one loads it in with rownames = geneID

provide the DE gene lists so that everyone starts with the same file.

#### Clustering

Is the gene expression normalized?  Doesn't seem like it...

#### Metagenomics

increase sample size?
Are the reps technical?  Prob not
Change the word "predominant" in the Qs


